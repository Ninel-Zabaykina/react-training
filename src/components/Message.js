import React from "react";
import PropTypes from "prop-types";

/**
 * Use `Message` to display chat messages
 */
const Message = function (props) {
    return <li data-testid="message">
        <strong>{props.message.nick}:</strong> {props.message.message}
    </li>;
};

Message.propTypes = {
    message: PropTypes.shape({
        nick: PropTypes.string,
        message: PropTypes.string
    }),
};

Message.defaultProps = {
    message: null,
};

export default Message;

