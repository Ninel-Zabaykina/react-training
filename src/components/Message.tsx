import React from "react";
import PropTypes from "prop-types";
import { IMessage} from "../interfaces";

type MessageProps = {
    message: PropTypes.InferProps<IMessage>;
};

/**
 * Use `Message` to display chat messages
 */
const Message: React.FC<MessageProps> = function (props: MessageProps) {
    return <li data-testid="message">
        <strong>{props.message.nick}:</strong> {props.message.message}
    </li>;
};

Message.propTypes = {
    message: PropTypes.shape({
        nick: PropTypes.string,
        message: PropTypes.string
    }).isRequired
};

export default Message;

