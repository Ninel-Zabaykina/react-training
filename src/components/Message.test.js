import React from 'react';
import { render } from '@testing-library/react';
import Message from './Message';

test('Component Message gets property message and displays it', () => {
    const message = {
        message: "Hello!",
        nick: "Nick"
    };
    const { container, getByText } = render(<Message message={message}/>);
    expect(getByText(/Nick/i)).toBeInTheDocument();
    expect(getByText(/Hello/i)).toBeInTheDocument();
});
