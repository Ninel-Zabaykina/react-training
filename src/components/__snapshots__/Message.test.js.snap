// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Message gets property message and displays it 1`] = `
<ul
  className="css-dnpncn-Message"
  data-testid="message"
>
  <strong
    className="css-sktysn-Message"
  >
    Nick
    :
  </strong>
  Â 
  Hello!
</ul>
`;
